### PR TITLE
Laravel 9.x Shift

### PR DESCRIPTION
**Pull request by Jellyfrog**
This pull request includes the changes for upgrading to Laravel 9.x. Feel free to commit any additional changes to the `shift-73269` branch.

**Before merging**, you need to:

- Checkout the `shift-73269` branch
- Review **all** pull request comments for additional changes
- Run `composer update` (if the scripts fail, add `--no-scripts`)
- Clear any config, route, or view cache
- Thoroughly test your application ([no tests?](https://laravelshift.com/laravel-test-generator), [no CI?](https://laravelshift.com/ci-generator))

If you need help with your upgrade, check out the [Human Shifts](https://laravelshift.com/human-shifts). You may also join the [Shifty Coders](https://laravelshift.com/shifty-coders) Slack workspace to level-up your Laravel skills.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
:information_source: Shift noticed you have additional namespaces in your application. You may use the [Consolidate Namespaces Shift](https://laravelshift.com/laravel-consolidate-custom-namespaces) to simplify your namespaces into the default Laravel `App` namespace.

**Comment by Jellyfrog** [comment]
:warning: In previous versions of Laravel, the `firstOrNew`, `firstOrCreate`, and `updateOrCreate` methods for a `belongsToMany` relationship compared against the pivot table for an existing record. In Laravel 9, they now compare against the related model.

Shift detected the following instances of these methods. You should review them to ensure you were not relying on the previous behavior.

- [ ] app/Http/Controllers/ServiceTemplateController.php

**Comment by Jellyfrog** [comment]
:information_source: Laravel 9 moved the `resources/lang` folder to the top level of the project. While Shift moved and replaced references to this folder, you may have additional references to this folder which need to be updated.

**Comment by Jellyfrog** [comment]
:x: Laravel 9 removed its dependency for `fruitcake/laravel-cors`. This behavior is now provided by the framework through `Illuminate\Http\Middleware\HandleCors`.

Shift found references to `Fruitcake\Cors` in the following files. You should review these references and adopt the internal middleware.

- [ ] app/Http/Middleware/HandleCors.php

**Comment by Jellyfrog** [comment]
:warning: Laravel 9 upgraded to Flysystem 3.0 causing the behavior of the `Storage` facade to change slightly:

* Write operations (such as `put` and `write`) now overwrite existing files by default, and now return `false` instead of an exception on failure.
* Read operations (such as `get` and `download`) now return `null` instead of a `FileNotFoundException` when a file is missing.
* The `delete` operation now returns `true` even if the file did not exist.

For more details, you may review the [Flysystem 3.x](https://laravel.com/docs/9.x/upgrade#flysystem-3) section of the _Upgrade Guide_.

**Comment by Jellyfrog** [comment]
:warning: Laravel 9 transitioned from SwiftMailer to Symfony Mailer. While this transition should be relatively seamless and Shift automates any tedious changes, there are some changes which are not reliably automated.

**Dependencies**
Some mail providers may require an additional dependency. For example, if you are using Mailgun, then you will need to install the `symfony/mailgun-mailer` package:

```sh
composer require symfony/mailgun-mailer symfony/http-client
```

**Updated Return Types**
The `send`, `html`, `text`, and `plain` methods no longer return the number of recipients that received the message.

**Proxied `Illuminate\Mail\Message` Methods**
Shift detected that you may be using calling some methods on `Illuminate\Mail\Message` that were being proxied to the underlying `Swift_Message` class. These are now being proxied to the `Symfony\Component\Mime\Email` class instead, which has different method names and signatures.

**Failed Recipients**
It is no longer possible to retrieve a list of failed recipients after sending a message. Instead, a `Symfony\Component\Mailer\Exception\TransportExceptionInterface` exception will be thrown if a message fails to send. Instead of relying on retrieving invalid email addresses after sending a message, it is recommend that you validate email addresses before sending the message instead.

For more details on what has changed you may review the [Symfony Mailer](https://laravel.com/docs/9.x/upgrade#symfony-mailer) section of the _Upgrade Guide_.

**Comment by Jellyfrog** [comment]
:x: Shift [automated the conversion](https://github.com/librenms/librenms/pull/14504/commits/2daba3fb706b75ccffbc8c1762692300283d292a?w=1) to fluent routes. However, some of your route groups contained unknown options. You should review the following files for any `group()` calls still passing an options array and manually convert them to their fluent method.

- [ ] routes/web.php

**Comment by Jellyfrog** [comment]
:warning: The `FILESYSTEM_DRIVER` environment variable was renamed to `FILESYSTEM_DISK` in Laravel 9. Shift attempted to [automate this change](https://github.com/librenms/librenms/pull/14504/commits/6c2ee5876e773bacc154a98565cc6d07e7ad3288?w=1), but you should review any additional environment configuration and rename this variable.

**Comment by Jellyfrog** [comment]
:warning: Shift upgraded your configuration files by defaulting them and merging your _true customizations_. These include values which are not changeable through core `ENV` variables.

You should [review this commit](https://github.com/librenms/librenms/pull/14504/commits/03a771bb537836e485f47b0abda9a82b1ec2e156?w=1) for additional customizations or opportunities to use new `ENV` variables. If you have a lot of customizations, you may undo this commit with `git revert 03a771bb` and make the [config file changes](https://github.com/laravel-shift/laravel-7.x/pull/2/commits/b6d066d9fa2b9482224f7b7332a3fa287e374924) manually.

**Comment by Jellyfrog** [comment]
:information_source: The `FILESYSTEM_CLOUD` environment variable was removed in a later release of Laravel 8. While you may still use the `cloud` disk, you are encouraged to [configure your own disks](https://laravel.com/docs/9.x/filesystem#configuration).

**Comment by Jellyfrog** [comment]
:warning: The `token` driver for API authentication is no longer available. This driver was not very robust and was removed from the documentation. Laravel now recommends using [Sanctum](https://laravel.com/docs/9.x/sanctum).

If you were using the `token` driver, you may re-add it within your `auth.php` configuration file then migrate to _Sanctum_ at your convenience.

**Comment by Jellyfrog** [comment]
:information_source: Shift [updated your dependencies](https://github.com/librenms/librenms/pull/14504/commits/6759aa08dc0e76d48421b016cb504e75eaf93bc8?w=1) for Laravel 9. While many of the popular packages are reviewed, you may have to update additional packages in order for your application to be compatible with Laravel 9. Watch [dealing with dependencies](https://laravelshift.com/videos/update-incompatible-composer-dependencies) for tips on handling any Composer issues.

The following dependencies were updated by a major version and may have their own changes. You may check their changelog for any additional upgrade steps.

- [ ] [laravel-notification-channels/webpush](https://packagist.org/packages/laravel-notification-channels/webpush)

**Comment by Jellyfrog** [comment]
:information_source: Laravel 9 adopted [anonymous migrations](https://laravel-news.com/laravel-anonymous-migrations). Shift [automated this change](https://github.com/librenms/librenms/pull/14504/commits/6251e1aee38b84d66c25f12cfd0e48297f022853?w=1) to align with modern Laravel conventions and avoid naming migrations.

**Comment by Jellyfrog** [comment]
:information_source: All of the underlying Symfony components used by Laravel have been upgraded to Symfony 6. If you are directly interacting with any Symfony component, you should review the [Symfony change log](https://github.com/symfony/symfony/blob/6.0/UPGRADE-6.0.md) for additional changes.

**Comment by Jellyfrog** [comment]
:warning: The `$loop` variable is no longer available when iterating over a `LazyCollection`. Shift detected the `$loop` variable being used in the following Blade templates. You should review these instances to ensure they are not iterating over a `LazyCollection` or track your own loop variables.

- [ ] resources/views/components/option-bar.blade.php
- [ ] resources/views/components/submenu.blade.php
- [ ] resources/views/device/tabs/latency.blade.php
- [ ] resources/views/device/tabs/vlans.blade.php
- [ ] resources/views/layouts/menu.blade.php
- [ ] resources/views/poller/poller.blade.php

**Comment by Jellyfrog** [comment]
:information_source: The HTTP client has been updated to have a default timeout of 30 seconds. Previously there was no default. The underlying request would wait indefinitely. Now an exception is thrown after 30 seconds.

If you find your requests need longer, you may prefix your request to specify a timeout:

```php
Http::timeout(60)->get("https://laravelshift.com");
```

**Comment by Jellyfrog** [comment]
:information_source: Laravel 9 now uses Vite to build frontend assets. While you may continue to use Laravel Mix, it is no longer the default. If you wish to modernize your application to use Vite, you may run the [Vite Converter](https://laravelshift.com/convert-laravel-mix-to-vite) for free.

**Comment by Jellyfrog** [comment]
:tada: Congratulations, you're now running the latest version of Laravel!

Next, you may optionally run the following _Shifts_ to ensure your application is fully upgraded, adopts the latest Laravel conventions, and easier to maintain in the future:

- [Upgrade Checker](https://laravelshift.com/laravel-upgrade-checker) ensures your application is fully upgraded by detecting any outdated code.
- [Laravel Fixer](https://laravelshift.com/laravel-code-fixer) automatically updates your code to the latest Laravel conventions.
- [Tests Generator](https://laravelshift.com/laravel-test-generator) intelligently generates model factories, HTTP Tests, and configuration for your application.
- [CI Generator](https://laravelshift.com/ci-generator) intelligently generates CI jobs to lint PHP, check code style, and run tests, including Dusk.

You may also use the [Shift Workbench](https://laravelshift.com/workbench) to automate common tasks for maintaining your Laravel and PHP applications.

File: config/cache.php
Comment (murrant) [inline config/cache.php:18]:
```suggestion
    'default' => env('CACHE_DRIVER', 'database'),
```

File: config/logging.php
Comment (murrant) [inline config/logging.php:76]:
```suggestion
            'level' => env('LOG_LEVEL', 'error'),
```

File: config/logging.php
Comment (murrant) [inline config/logging.php:83]:
```suggestion
            'level' => env('LOG_LEVEL', 'error'),
```

File: config/logging.php
Comment (murrant) [inline config/logging.php:110]:
```suggestion
            'formatter' => env('LOG_STDERR_FORMATTER', \App\Logging\CliColorFormatter::class),
```

File: config/session.php
Comment (murrant) [inline config/session.php:34]:
```suggestion
    'lifetime' => env('SESSION_LIFETIME', 34560),
```

File: routes/web.php
Comment (Jellyfrog) [inline routes/web.php:31]: @murrant had to remove the `'guard' => 'auth'` for it to work, not sure if this is needed, there is no guard named auth?

**Comment by Jellyfrog** [comment]
Blocked by #14526

**Comment by Jellyfrog** [comment]
- [ ] Add https://github.com/librenms/librenms/commit/cf9796b74191b834fa399ced06ace05f66557699 as separate PR

**Comment by Jellyfrog** [comment]
- [ ] Enable phpstan level 6

**Comment by murrant** [comment]
Why are tests taking 1.5hrs?

**Comment by Jellyfrog** [comment]
@murrant should we add `$this->expectNotToPerformAssertions();` to these ?

```
Time: 33:27.404, Memory: 842.50 MB

There were 7 risky tests:

1) LibreNMS\Tests\Unit\Data\GraphiteStoreTest::testSocketConnectError
This test did not perform any assertions

librenms/tests/Unit/Data/GraphiteStoreTest.php:55

2) LibreNMS\Tests\Unit\Data\GraphiteStoreTest::testSocketWriteError
This test did not perform any assertions

librenms/tests/Unit/Data/GraphiteStoreTest.php:65

3) LibreNMS\Tests\Unit\Data\GraphiteStoreTest::testSimpleWrite
This test did not perform any assertions

librenms/tests/Unit/Data/GraphiteStoreTest.php:76

4) LibreNMS\Tests\Unit\Data\OpenTSDBStoreTest::testSocketConnectError
This test did not perform any assertions

librenms/tests/Unit/Data/OpenTSDBStoreTest.php:55

5) LibreNMS\Tests\Unit\Data\OpenTSDBStoreTest::testSocketWriteError
This test did not perform any assertions

librenms/tests/Unit/Data/OpenTSDBStoreTest.php:65

6) LibreNMS\Tests\Unit\Data\OpenTSDBStoreTest::testSimpleWrite
This test did not perform any assertions

librenms/tests/Unit/Data/OpenTSDBStoreTest.php:76

7) LibreNMS\Tests\Unit\Data\OpenTSDBStoreTest::testPortWrite
This test did not perform any assertions

librenms/tests/Unit/Data/OpenTSDBStoreTest.php:93
```

File: routes/web.php
Comment (Jellyfrog) [reply routes/web.php:31]: This is left to check

Comment (Jellyfrog) [inline routes/web.php:31]: TODO:

- [ ] Fix this one

**Comment by Jellyfrog** [comment]
This gets logged which it didn't previously:
`[2022-11-16T15:33:32.392966+00:00] production.ERROR: No Processors {"userId":1,"exception":"[object] (LibreNMS\Exceptions\RrdGraphException(code: 0): No Processors at librenms/includes/html/graphs/device/processor.inc.php:6`

File: app/Http/Requests/TwoFactorManagementRequest.php
Comment (Jellyfrog) [inline app/Http/Requests/TwoFactorManagementRequest.php:43]: This should be correct, right? Phpstan whine

File: routes/web.php
Comment (murrant) [reply routes/web.php:31]: Double done.

**Review by murrant — APPROVED** [review]

**Pull request merged**
